tree/ntuple/v7/inc/ROOT/RMiniFile.hxx

```diff
@@ -0,0 +1,221 @@
+/// \file ROOT/RMiniFile.hxx
+/// \ingroup NTuple ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2019-12-22
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RMiniFile
+#define ROOT7_RMiniFile
+
+#include <ROOT/RNTupleOptions.hxx>
+#include <ROOT/RStringView.hxx>
+
+#include <cstdint>
+#include <cstdio>
+#include <memory>
+#include <string>
+
+class TFile;
+
+namespace ROOT {
+namespace Experimental {
+
+namespace Detail {
+class RRawFile;
+}
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTuple
+\ingroup NTuple
+\brief Entry point for an RNTuple in a ROOT file
+
+The class points to the header and footer keys, which in turn have the references to the pages.
+Only the RNTuple key will be listed in the list of keys. Like TBaskets, the pages are "invisible" keys.
+Byte offset references in the RNTuple header and footer reference directly the data part of page records,
+skipping the TFile key part.
+*/
+// clang-format on
+struct RNTuple {
+   /// Allows for evolving the struct in future versions
+   std::uint32_t fVersion = 0;
+   /// Allows for skipping the struct
+   std::uint32_t fSize = sizeof(RNTuple);
+   /// The file offset of the header excluding the TKey part
+   std::uint64_t fSeekHeader = 0;
+   /// The size of the compressed ntuple header
+   std::uint32_t fNBytesHeader = 0;
+   /// The size of the uncompressed ntuple header
+   std::uint32_t fLenHeader = 0;
+   /// The file offset of the footer excluding the TKey part
+   std::uint64_t fSeekFooter = 0;
+   /// The size of the compressed ntuple footer
+   std::uint32_t fNBytesFooter = 0;
+   /// The size of the uncompressed ntuple footer
+   std::uint32_t fLenFooter = 0;
+   /// Currently unused, reserved for later use
+   std::uint64_t fReserved = 0;
+
+   bool operator ==(const RNTuple &other) const {
```

Comment:
If this is different from the implicit operator==, it should be noted here (humm is there even an implicit operator==?)

Reply:
Unfortunately there is no implicit `==` operator.  I added a clarifying comment.